pkg/apis/serving/v1alpha1/cluster_local_model_validation.go

```diff
@@ -0,0 +1,91 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	// logger for the validation webhook.
+	clusterLocalModelValidatorLogger = logf.Log.WithName("clusterlocalmodel-v1alpha1-validation-webhook")
+)
+
+// +kubebuilder:object:generate=false
+// +k8s:deepcopy-gen=false
+// +k8s:openapi-gen=false
+// ClusterLocalModelValidator is responsible for validating the ClusterLocalModelValidator resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false and +k8s:deepcopy-gen=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type ClusterLocalModelValidator struct{}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-clusterlocalmodels,mutating=false,failurePolicy=fail,groups=serving.kserve.io,resources=clusterlocalmodels,versions=v1alpha1,name=clusterlocalmodel.kserve-webhook-server.validator
+var _ webhook.CustomValidator = &ClusterLocalModelValidator{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (v *ClusterLocalModelValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	clusterLocalModel, err := convertToClusterLocalModel(obj)
+	if err != nil {
+		clusterLocalModelValidatorLogger.Error(err, "Unable to convert object to ClusterLocalModel")
+		return nil, err
+	}
+	clusterLocalModelValidatorLogger.Info("validate create", "name", clusterLocalModel.Name)
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (v *ClusterLocalModelValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	clusterLocalModel, err := convertToClusterLocalModel(newObj)
+	if err != nil {
+		clusterLocalModelValidatorLogger.Error(err, "Unable to convert object to ClusterLocalModel")
+		return nil, err
+	}
+	clusterLocalModelValidatorLogger.Info("validate update", "name", clusterLocalModel.Name)
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (v *ClusterLocalModelValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	clusterLocalModel, err := convertToClusterLocalModel(obj)
+	if err != nil {
+		clusterLocalModelValidatorLogger.Error(err, "Unable to convert object to ClusterLocalModel")
+		return nil, err
+	}
+	clusterLocalModelValidatorLogger.Info("validate delete", "name", clusterLocalModel.Name)
+
+	// Check if current ClusterLocalModel is being used
+	// Check the status of each isvc in clusterLocalModel.Status.InferenceServices?
```

Comment:
if possible, could we list all isvcs then check if label matches?
https://github.com/kserve/kserve/blob/edcd1ef72c772b222c94727882bdbcdd4ea29c1d/pkg/controller/v1alpha1/localmodel/controller.go#L308
otherwise just check if `clusterLocalModel.Status.InferenceServices` is empty.